### DISCUSSION TITLE
Copilot does not understand code from external files, but there is a way to do it

### DISCUSSION THREAD

**Discussion by oxysoft** [community/community#17637] [Copilot Conversations]
Hi, I use Copilot in Jetbrains Rider with C#. I have moderate success. I have noticed that Copilot doesn't understand code that is being referenced nearby my caret or classes in other files, it does not understand the fields and functions in those classes, etc.

A solution to this problem is to look at what types or functions would be useful for a completion, and concatenate the relevant content of those files when send to the AI. That way, it's now part of the context.

What do you guys think?

**Comment by ZayshaaCodes** [comment]
Just providing it with a basic class structure with public members for relevant types would do wonders. would save a lot of tokens

**Comment by samdus** [comment]
It would also prevent Copilot from autocompleting code that uses methods that don't exist in the external class. (I have the same problem with Copilot in IntelliJ with Java.)

**Reply by oxysoft** [reply]
Hmmmmmmm it is a pretty basic feature to have in copilot, yet it doesn't seem likely to come... weird. Copilot seems behind in terms of innovation with what LLMs can do today, open-source could take over soon.
